app/services/mhv_accounts_service.rb

```diff
@@ -42,60 +41,57 @@ def create
   end
 
   def upgrade
-    if mhv_account.may_upgrade?
-      client_response = mhv_ac_client.post_upgrade(params_for_upgrade)
-      if client_response[:status] == 'success'
-        StatsD.increment("#{STATSD_ACCOUNT_UPGRADE_KEY}.success")
-        mhv_account.upgraded_at = Time.current
-        mhv_account.upgrade!
-      end
-    end
-  rescue => e
-    if e.is_a?(Common::Exceptions::BackendServiceException) && e.original_body['code'] == 155
-      StatsD.increment(STATSD_ACCOUNT_EXISTED_KEY.to_s)
-      mhv_account.upgrade! # without updating the timestamp since account was not created at vets.gov
+    if mhv_account.upgradable?
+      handle_upgrade!
+    elsif mhv_account.account_level == 'Error'
+      log_message_to_sentry('Possible Race Condition In MHV Upgrade', :warn, extra_context: mhv_account.attributes)
```

Comment:
In this case, it seems like the controller would render a status of accepted. Not sure if that is the appropriate response for when this happens. Does this need to raise an exception?

Reply:
Hmmm, that's a good point. I'm thinking we would want to raise an exception saying that we cannot upgrade your account at this time, try again later, maybe?